nncf/torch/quantization/init_range.py

```diff
@@ -145,7 +160,7 @@ def generate_stat_collector_for_range_init_config(
 
     @staticmethod
     def get_all_scale_shapes(qp: QuantizationPointBase,
-                             target_nncf_graph: PTNNCFGraph) -> List[Tuple[int]]:
+                             target_nncf_graph: PTNNCFGraph) -> List[Tuple[Tuple[int], QuantizationMode, bool]]:
```

Comment:
maybe returning of the class with properties would be helpful here. you could pass this structure further to collectors without parsing

Reply:
Thanks for the comment. It seems that in this case, the simplest scenario is just to create `rs_vs_params` dictionary object right within the `get_all_scale_shapes_with_params` and return it. This is simple enough and helps to avoid additional structures.